Library/Homebrew/formula_cellar_checks.rb

```diff
@@ -56,14 +56,19 @@ def check_jars
     EOS
   end
 
+  VALID_LIBRARY_EXTENSIONS = %w[.a .dylib .framework .jnilib .la .o .so .jar .prl .pm .sh].freeze
+
+  def valid_library_extension?(filename)
+    VALID_LIBRARY_EXTENSIONS.include? filename.extname
```

Comment:
Apologies for a late comment. Would it make sense to define `VALID_OS_SPECIFIC_LIBRARY_EXTENSIONS` variable and then do 
```ruby
(VALID_LIBRARY_EXTENSIONS + VALID_OS_SPECIFIC_LIBRARY_EXTENSIONS).include? filename.extname
```
?

Reason: `.dylib` and `.framework` are not used as library extensions on Linux.

Reply:
Good point. Within the current framework, `extend/os/mac/formula_cellar_checks.rb` should override `valid_library_extension?` and `.dylib .framework` should be moved to this macOS-specific function, similar to how Linux handles `*.so.*` in `extend/os/linux/formula_cellar_checks.rb`. Would you like to open that PR?

Reply:
> Would you like to open that PR?

1. PR against Linuxbrew first?
1. Can I get couple of extra hours in a day to go please? :)

Reply:
Hah! For tricky PRs, I'd suggest Linuxbrew first, but this one is pretty straight forward, so a PR direct to Homebrew/brew ought to be fine. Tag me, and I'll review.